client-src/elements/chromedash-header_test.ts

```diff
@@ -32,9 +32,9 @@ describe('chromedash-header', () => {
 
     // invalid feature requests would trigger the toast to show message
     const toastEl = document.querySelector('chromedash-toast');
-    const toastMsgSpan = toastEl.shadowRoot.querySelector('span#msg');
+    const toastMsgSpan = toastEl?.shadowRoot?.querySelector('span#msg');
```

Comment:
Should this one reference `renderRoot` as well?  
What's the rule for choosing shadowRoot vs renderRoot?

Reply:
choosing renderRoot over shadowRoot is because if we assert instance it'll automatically figure out that component.renderRoot isn't null, so we dont have to add `!` or `?`. However, for this one we're accesing shadowDOM of an element that's outside of the class so renderRoot is not applicable